src/thor/timedep_forward.cc

```diff
@@ -202,8 +268,9 @@ inline bool TimeDepForward::ExpandForwardInner(GraphReader& graphreader,
     // a path to be formed even if the convergence test fails (can
     // happen with large edge scores)
     if (best_path.first == -1 || newcost.cost < best_path.second) {
-      best_path.first = (meta.edge_status->set() == EdgeSet::kTemporary) ? meta.edge_status->index()
-                                                                         : edgelabels_.size();
+      best_path.first = (meta.edge_status->set() == EdgeSet::kTemporary)
+                            ? meta.edge_status->index()
+                            : (FORWARD ? edgelabels_.size() : edgelabels_.size());
```

Comment:
Note: these used to be different (`edgelabels_rev_` and `edgelabels_`) - I've left this in for now as a reminder that forward/reverse behaviour may be specialized here.